python/paddle/amp/auto_cast.py

```diff
@@ -212,7 +257,9 @@ def need_keep_fp32(layer, dtype):
     return need_keep_fp32
 
 
-def set_excluded_layers(models, excluded_layers):
+def set_excluded_layers(
+    models: Layer | list[Layer], excluded_layers: Layer | list[Layer]
```

Comment:
```suggestion
    models: Layer | list[Layer], excluded_layers: Layer | list[Layer] | list[type[Layer]]
```

后面

``` python
            elif issubclass(item, paddle.nn.Layer):
                excluded_layers_types.append(item)
```

应该还需要 `type[Layer]` ~